### PR TITLE
PR: Add methods to simplify access to our config system in `MainWindow`

### PR DESCRIPTION
**Pull request by ccordoba12**
## Description of Changes

- Small refactoring whose main purpose is to add `get_conf`/`set_conf` methods to MainWindow in order to simplify how we access config options on that class.
- Move `__init__` to be at the beginning of MainWindow (I think it's odd to have it in the middle of the class).
- Move the Qt methods we override on MainWindow to be on their own section.
- Add more section separators to MainWindow to make easier to navigate it.
- Remove old Mac stylesheet because it's no longer needed.

*Notes*:
1. I did these changes while working to fix another bug. I preferred to break that into two parts to facilitate reviewing it.
3. I think we should avoid using `CONF` because it's cleaner and easier to read to have methods that call it behind the scenes.

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @ccordoba12

### CODE REVIEW

**Comment by ccordoba12** [comment]
Let's hold this PR until PR #18778 is merged because it touches the same code.

**Comment by ccordoba12** [comment]
@dalthviz, scratch my last comment because Ryan agreed to leave PR #18778 for Spyder 6.

File: spyder/app/mainwindow.py
Comment (dalthviz) [inline spyder/app/mainwindow.py:746]: Maybe a new method call `get_conf_shortcut` should be implemented which returns `self._conf.get_shorcut` in a similar way as done for `set_conf` and `get_conf`?

**Review by dalthviz — COMMENTED** [review]
Thanks @ccordoba12 ! Left a comment regarding the creation of a method to access shortcuts in a similar manner as to get and set configurations. Other than that this LGTM :+1:

**Review by dalthviz — APPROVED** [review]
:+1:

**Pull request merged**
